### PR TITLE
Add detection of invalid security sensitive config

### PR DESCRIPTION
**Pull request by wendigo**
cc @findepi 

to prevent situations like https://github.com/trinodb/trino/pull/19519

### CODE REVIEW

File: bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java [hunk 1]
```diff
@@ -67,6 +67,8 @@ public class Bootstrap
 
     private Map<String, String> requiredConfigurationProperties;
     private Map<String, String> optionalConfigurationProperties;
+    private List<String> securitySensitivePatterns;
```
Comment (kokosing) [inline bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java:70]: Is pattern a regex? Below it looks like you are using `contains`. Can we have a good default list? So one does not need to configure whenever `Boostrap` is used?

**Review by kokosing — COMMENTED** [review]
Nice!

File: bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java (hunk 1 above)
Comment (wendigo) [reply bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java:70]: @kokosing it's as substring, not a pattern, I could make it a regex

File: bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java [hunk 2]
```diff
@@ -119,6 +121,15 @@ public Bootstrap setOptionalConfigurationProperties(Map<String, String> optional
         return this;
     }
 
+    public Bootstrap setSensitiveConfigPatterns(List<String> patterns)
+    {
+        if (this.securitySensitivePatterns == null) {
+            this.securitySensitivePatterns = new ArrayList<>();
+        }
+        this.securitySensitivePatterns.addAll(patterns);
```
Comment (findepi) [inline bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java:129]: setSensitiveConfigPatterns -> add....

File: bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java [hunk 3]
```diff
@@ -119,6 +121,15 @@ public Bootstrap setOptionalConfigurationProperties(Map<String, String> optional
         return this;
     }
 
+    public Bootstrap setSensitiveConfigPatterns(List<String> patterns)
+    {
+        if (this.securitySensitivePatterns == null) {
+            this.securitySensitivePatterns = new ArrayList<>();
```
Comment (findepi) [inline bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java:127]: why not initialize the list directly on the field?

File: bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java [hunk 4]
```diff
@@ -119,6 +121,15 @@ public Bootstrap setOptionalConfigurationProperties(Map<String, String> optional
         return this;
     }
 
+    public Bootstrap setSensitiveConfigPatterns(List<String> patterns)
```
Comment (findepi) [inline bootstrap/src/main/java/io/airlift/bootstrap/Bootstrap.java:124]: unused?

File: configuration/src/test/java/io/airlift/configuration/TestConfig.java [hunk 5]
```diff
@@ -204,6 +205,33 @@ public void testDetectsNoConfigAnnotations()
         }
     }
 
+    @Test
+    public void testDetectsSecuritySensitiveConfig()
+    {
+        try {
+            Injector injector = createInjector(Collections.<String, String>emptyMap(), ImmutableList.of("access-key"), createModule(Key.get(SecuritySensitiveConfig.class), SecuritySensitiveConfig.class, null));
```
Comment (findepi) [inline configuration/src/test/java/io/airlift/configuration/TestConfig.java:212]: I understand that this provides a form of a unit test that the annotation is there.
However, that means i either need to remember to add annotation or to add a test (and then add annotation), which relies on remembering.

so the add value of this is that, once i have a test, one cannot remove the annotation without updating the test

**Comment by findepi** [comment]
if i am reading this correctly, then I don't understand how it helps find configs missing security sensitive annotation.
how does it work?

**Comment by wendigo** [comment]
@findepi this will fail on runtime while constructing config with keys that are not annotated but match patterns

**Pull request closed without merging**